### PR TITLE
Focused title implementation to listview

### PR DESCRIPTION
**Pull request by TomaszLukawskiSam**
[Issue] https://github.com/Samsung/TAU/issues/678
[Problem] Title Bar - Focused - is not implemented
[Solution]
 - examples in UIComponents according to guidelines
 - added top overscroll for ArcListview
 - scroll header to center position

Signed-off-by: Tomasz Lukawski <t.lukawski@samsung.com>

### CODE REVIEW

**Comment by TomaszLukawskiSam** [comment]
This PR partially solved other issue: https://github.com/Samsung/TAU/issues/679

File: src/js/profile/wearable/widget/wearable/ArcListview.js [hunk 1]
```diff
@@ -267,7 +272,14 @@
 
 				// virtual list parameters
 				NUMBER_ITEMS_TO_ADD = 20,
-				LOAD_THRESHOLD = 10;
+				LOAD_THRESHOLD = 10,
+				overscrollRotaryHandler = null,
```
Comment (grzegorz-czajkowski) [inline src/js/profile/wearable/widget/wearable/ArcListview.js]: I think we should move this out from constants.

File: src/js/profile/wearable/widget/wearable/ArcListview.js [hunk 2]
```diff
@@ -1029,11 +1068,13 @@
 			prototype._onRotary = function (event) {
 				var self = this;
 
-				self._scrollAnimationEnd = true;
-				if (event.detail.direction === "CW") {
-					self._rollDown();
-				} else {
-					self._rollUp();
+				if (!self._ui.header.classList.contains(classes.HEADER_FOCUSED)) {
```
Comment (grzegorz-czajkowski) [inline src/js/profile/wearable/widget/wearable/ArcListview.js]: Shouldn't we check if header exists?

File: src/js/profile/wearable/widget/wearable/ArcListview.js [hunk 3]
```diff
@@ -1234,6 +1273,41 @@
...
+				if (self._scrollAnimationEnd) {
+					state.startTime = Date.now();
+					self._scrollAnimationEnd = false;
+					self._requestRender();
+				}
+
+				// clear timeout
+				if (overscrollRotaryHandler) {
+					window.clearTimeout(overscrollRotaryHandler);
+					overscrollRotaryHandler = 0;
```
Comment (grzegorz-czajkowski) [inline src/js/profile/wearable/widget/wearable/ArcListview.js]: null instead of ?

File: src/js/profile/wearable/widget/wearable/ArcListview.js [hunk 4]
```diff
@@ -1234,6 +1273,41 @@
...
+
+				// if scroll animation is ended then animation start
+				if (self._scrollAnimationEnd) {
+					state.startTime = Date.now();
+					self._scrollAnimationEnd = false;
+					self._requestRender();
+				}
+
+				// clear timeout
+				if (overscrollRotaryHandler) {
```
Comment (grzegorz-czajkowski) [inline src/js/profile/wearable/widget/wearable/ArcListview.js]: Wouldn't be better to have overscrollRotaryHandler as member of ArcListview?

**Comment by grzegorz-czajkowski** [comment]
I've verified this PR. Works fine.

File: src/js/profile/wearable/widget/wearable/ArcListview.js (hunk 1 above)
Comment (TomaszLukawskiSam) [reply src/js/profile/wearable/widget/wearable/ArcListview.js]: Done

File: src/js/profile/wearable/widget/wearable/ArcListview.js (hunk 2 above)
Comment (TomaszLukawskiSam) [reply src/js/profile/wearable/widget/wearable/ArcListview.js]: Done

File: src/js/profile/wearable/widget/wearable/ArcListview.js (hunk 4 above)
Comment (TomaszLukawskiSam) [reply src/js/profile/wearable/widget/wearable/ArcListview.js]: Done

File: src/js/profile/wearable/widget/wearable/ArcListview.js (hunk 3 above)
Comment (TomaszLukawskiSam) [reply src/js/profile/wearable/widget/wearable/ArcListview.js]: Done

**Review by grzegorz-czajkowski — APPROVED** [review]
Looks good to me.

**Pull request merged**